draft-ietf-taps-interface.md

```diff
@@ -705,6 +706,65 @@ ChallengeCallback := NewCallback({
 SecurityParameters.SetIdentityChallengeCallback(challengeCallback)
 ~~~
 
+## Specifying Caching Context {#cache-context}
+
+During the lifetime of Connection, the various protocols used may generate 
+state or statistics that can be cached to help optimize future Connections.
+New Connections can then take advantage of this cached state to influence
+connection establishment, tuning protocol heuristics, and resuming state.
+
+Some examples of cached state include:
+
+- DNS query answers may be cached for use in subsequent Connections
+to the same hostname.
+
+- TCP round-trip-time (RTT) metrics may be cached to tune delays for
+protocol stack racing.
+
+- TLS state may be cached for use in session resumption.
+
+Within a single application, there may need to be several distinct and separate
+caches to ensure that data and state is not shared between certain Connections.
+Each Preconnection is therefore associated with a CacheContext, which defines
+a distinct set of cached information. All Connections and Preconnections that are
+associated with a single CacheContext are allowed to share state.
```

Comment:
I'd piggyback this with a blurb similar to the above, wherein we only want use that shared state if necessary.